react/src/components/LegacyFolderExplorer.tsx

```diff
@@ -89,10 +133,82 @@ const LegacyFolderExplorer: React.FC<LegacyFolderExplorerProps> = ({
         <Flex justify="between" gap={token.marginMD} style={{ width: '100%' }}>
           <Flex gap={token.marginMD} style={{ flex: 1 }}>
             <Suspense fallback={<Skeleton.Input active />}>
-              <VFolderNameTitle vfolderID={vfolderID} />
+              <VFolderNameTitle
+                vfolderFrgmt={vfolder ?? undefined}
+                vfolderNodeFrgmt={vfolder_node ?? undefined}
+              />
             </Suspense>
           </Flex>
-          <Flex justify="end" gap={token.marginSM} style={{ flex: lg ? 2 : 1 }}>
+          <Flex
+            justify="end"
+            gap={token.marginSM}
+            style={{
+              flex: lg ? 2 : 1,
+            }}
+          >
+            <BAISelect
+              defaultValue={vfolder_node?.permission || vfolder?.permission}
+              options={[
+                { value: 'ro', label: t('data.ReadOnly') },
+                { value: 'rw', label: t('data.ReadWrite') },
+              ]}
+              onChange={(value) => {
+                updateMutation.mutate(
+                  { permission: value, id: vfolderID },
+                  {
+                    onSuccess: () => {
+                      message.success(t('data.permission.PermissionModified'));
+                    },
+                    onError: (error) => {
+                      message.error(painKiller.relieve(error?.message));
+                    },
+                  },
+                );
+              }}
+              popupMatchSelectWidth={false}
+            />
+            <Button
+              icon={
+                <Image
+                  width="18px"
+                  src="/resources/icons/filebrowser.svg"
+                  alt="File Browser"
+                  preview={false}
+                />
+              }
+              onClick={() =>
+                // @ts-ignore
+                folderExplorerRef.current?._executeFileBrowser()
+              }
+            >
+              {lg && t('data.explorer.ExecuteFileBrowser')}
+            </Button>
+            <Button
+              icon={
+                <Image
+                  width="18px"
+                  src="/resources/icons/sftp.png"
+                  alt="SSH / SFTP"
+                  preview={false}
+                />
+              }
+              onClick={() => {
+                // @ts-ignore
+                folderExplorerRef.current?._executeSSHProxyAgent();
+              }}
+            >
+              {lg && t('data.explorer.RunSSH/SFTPserver')}
+            </Button>
+          </Flex>
+          <Flex
+            justify="end"
+            gap={token.marginSM}
+            style={{
+              position: 'absolute',
```

Comment:
This might cause  encroaches button field by breadcrumb. 

This can be avoided by setting max-width: 60% for the div.breadcrumb inside `backend-ai-folder-explorer` and padding: 0 setting to avoid this.